addon/model/model-text.ts

```diff
@@ -28,24 +27,28 @@ export default class ModelText extends ModelNode {
     return this._content.length;
   }
 
-  get textAttributeMap(): Map<TextAttribute, boolean> {
-    return this._textAttributeMap;
+  get isBlock() {
+    return false;
   }
 
-  set textAttributeMap(value: Map<TextAttribute, boolean>) {
-    this._textAttributeMap = value;
+  get offsetSize() {
+    return this.length;
```

Comment:
should we consider odd cases like "   abc  " here (which has only 4 visible positions, starting right before the a and ending right after c).

Reply:
no it has 6 visible positions: 
before: space,a,b,c,space,end

Reply:
note that's not true for real text nodes though, " abc " renders as "abc", per https://developer.mozilla.org/en-US/docs/Web/API/Document_Object_Model/Whitespace

Reply:
oh good god

Reply:
That sounds like a plan, we should take care though when normalizing. Take splitting text nodes for example. If the resulting nodes start or end with a space we probably want to convert that a non breaking space (if we don't already do that) so it shows up. (e.g. split "I walk my dog" into "I walk&nbsp;" "my dog"). on normalizing I suppose we want to convert that non breaking space back to a normal space so we have "I walk my dog" instead of "I walk&nbsp;my dog")